### PR TITLE
[conn] Add two AST related connectivity tests

### PR DESCRIPTION
**Pull request by cindychip**
This PR adds two connectivity tests between ast and flash, ast and
spi_device.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

File: hw/top_earlgrey/formal/conn_csvs/flash_ast.csv [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Run these checks with:
+#  ./util/dvsim/dvsim.py hw/top_earlgrey/formal/chip_conn_cfg.hjson
+
+,NAME,SRC BLOCK,SRC SIGNAL,DEST BLOCK,DEST SIGNAL,,,,,,
+
+CONNECTION,FLASH_AST,top_earlgrey.u_flash_ctrl,obs_ctrl_i,u_ast,obs_ctrl_o
```
Comment (matutem) [inline hw/top_earlgrey/formal/conn_csvs/flash_ast.csv:10]: Nit: Could you reverse the order to be src -> dest, so output -> input.

File: hw/top_earlgrey/formal/conn_csvs/spi_device_ast.csv [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Run these checks with:
+#  ./util/dvsim/dvsim.py hw/top_earlgrey/formal/chip_conn_cfg.hjson
+
+,NAME,SRC BLOCK,SRC SIGNAL,DEST BLOCK,DEST SIGNAL,,,,,,
+
+CONNECTION,SPI_DEVICE_AST,top_earlgrey.u_spi_device,ram_cfg_i,u_ast,"{dpram_rmf_o, dpram_rml_o}"
```
Comment (matutem) [inline hw/top_earlgrey/formal/conn_csvs/spi_device_ast.csv:10]: Same as above.

**Review by matutem — APPROVED** [review]
With a couple minor nit, LGTM.

File: hw/top_earlgrey/formal/conn_csvs/flash_ast.csv (hunk 1 above)
Comment (cindychip) [reply hw/top_earlgrey/formal/conn_csvs/flash_ast.csv:10]: Thanks Matute! Will update that.

File: hw/top_earlgrey/formal/conn_csvs/spi_device_ast.csv (hunk 2 above)
Comment (cindychip) [reply hw/top_earlgrey/formal/conn_csvs/spi_device_ast.csv:10]: Thanks again!

**Pull request merged**